src/statement.d

```diff
@@ -1421,6 +1421,10 @@ public:
                     statements.insert(i, flt);
                     continue;
                 }
+                // Allow CommaExp in ExpStatement because return isn't used
```

Comment:
It may be worth encapsulating this check-and-set pattern into a function, eg `discardCommaResult` that walks the expression tree if necessary.